source/common/common/assert.h

```diff
@@ -3,21 +3,34 @@
 #include "common/common/logger.h"
 
 namespace Envoy {
+
 /**
  * assert macro that uses our builtin logging which gives us thread ID and can log to various
  * sinks.
+ *
+ * The old style release assert was of the form RELEASE_ASSERT(foo == bar);
+ * where it would log stack traces and the failed conditional and crash if the
+ * condition is not met.  The are many legacy RELEASE_ASSERTS in Envoy which
+ * were converted to RELEASE_ASSERT(foo == bar, "");
+ *
+ * The new style of release assert is of the form
+ * RELEASE_ASSERT(foo == bar, "reason foo should actually be bar");
+ * new uses of RELEASE_ASSERT are strongly encouraged to supply a verbose
+ * explanation of what went wrong.
  */
-#define RELEASE_ASSERT(X)                                                                          \
+#define RELEASE_ASSERT(X, DETAILS)                                                                 \
```

Comment:
Do you want `...` and `__VA_ARGS__` here to support arbitrary fmtlib?

Reply:
I figured we'd have folks do fmt:: inline as I did in the unit tests.

Reply:
I think it's like a one line change in the macro to have it do the magic for the user (and default to something sane), e.g. (untested):
```
#define RELEASE_ASSERT(X, DETAILS,...) \
...
  ENVOY_LOG_TO_LOGGER(Envoy::Logger::Registry::getLog(Envoy::Logger::Id::assert), critical,  "assert failure: {}.{}{}", #X, details.empty() ? "" : " Details: ", fmt::format(details, ##__VA_ARGS__); \
...
```
see https://gcc.gnu.org/onlinedocs/cpp/Variadic-Macros.html

Reply:
I'm not convinced it's that simple.  It could be you have some minor typo, but adding in the vargs as you proposed lands me an 
assert error argument index out of range
Meanwhile I find 
{ RELEASE_ASSERT(0 == EAGAIN, fmt::format("using {}", "fmt")); },
more readable than
{ RELEASE_ASSERT(0 == EAGAIN, "native {}", "fmt"); },
because it's not inherently a logging macro.